### PR TITLE
Hallucinations part 2: Chem Factor

### PR DESCRIPTION
**Pull request by Doxxmedearly**
- tweak: "Synaptizine, methylphenidate, risperidone and olanzapine will reduce overall hallucination duration, reduce frequency of hallucination effects, and sometimes prevent them entirely. Useful!"
  - tweak: "Some hallucinogenic drugs and toxins will now increase the duration you hallucinate and make effects happen more frequently."
  - tweak: "Wulumunusha will now cause skrell to experience light hallucinations (as its description implies) at dosages greater than 10u."
  - bugfix: "Fixed null hallucinations causing runtimes. Hopefully."

Adds the CE_MIND chem effect, which basically makes hallucinations easier to deal with (less duration, less effects) or harder (longer duration, frequent effects). Adds it to the appropriate reagents. Everything else as indicated on the CL.

### CODE REVIEW

**Comment by Doxxmedearly** [comment]
!review

File: code/controllers/subsystems/hallucinations.dm [hunk 1]
```diff
@@ -29,6 +29,9 @@
 	var/list/candidates = list()
 	for(var/T in all_hallucinations)
 		var/datum/hallucination/H = new T
+		if(isnull(H))
+			log_debug("Null hallucination found while creating new in subsystem.")
+			continue
```
Comment (NonQueueingMatt) [inline code/controllers/subsystems/hallucinations.dm:34]: This seems like a weird thing to add? If you're using subtypesof you should like never get a null hallucination out of that. Something is weird elsewhere.

Comment (Doxxmedearly) [reply code/controllers/subsystems/hallucinations.dm:34]: Thinking on it, yeah. The isnull() in handle_hallucination should be enough. I'll remove it when I can.

**Comment by Doxxmedearly** [comment]
Might reverse it and fix the names, then. I couldn't figure out a good rename to CE_MIND that didn't fuck with what's coded, but yeah I'm not happy with it either. Will probably go in and fix.

**Comment by Doxxmedearly** [comment]
Uno reversed.
ANTIHALLUCINATE is now HALLUCINATE. Beneficial drugs add negative CE, harmful ones add regular CE. Should make more sense naming-wise.

**Pull request merged**